plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/submitit_launcher.py

```diff
@@ -34,13 +34,16 @@ def manipulate_search_path(self, search_path: ConfigSearchPath) -> None:
         )
 
 
-class SubmititLauncher(Launcher):
+class LocalSubmititLauncher(Launcher):
+
+    _EXECUTOR = "local"
+
     def __init__(self, **params: Any) -> None:
         self.params = OmegaConf.structured(
-            LocalParams if params["executor"].value == "local" else SlurmParams
+            LocalParams if self._EXECUTOR == "local" else SlurmParams
         )
```

Comment:
I suggest adding a base class to express that the slurm launcher is not REALLY a local launcher.

```python
# your current local:
class BaseSubmititLauncher(Launcher):
  def _get_params_class() -> Type[BaseParams]:
     raise NotImplementedError

  ... # the rest is the same.

class LocalSubmititLauncher(BaseSubmititLauncher)
  def _get_params_class() -> Type[BaseParams]:
     return LocalParams 

class SlurmSubmititLauncher(BaseSubmititLauncher)
  def _get_params_class() -> Type[BaseParams]:
     return SlurmParams
```

You can go back to a DictConfig with something like this (and now you don't need the update loop)
```python
self.params = OmegaConf.structured(self._get_params_class(**params))
```